examples/ethernet/ethernet/main/ethernet_main.c

```diff
@@ -149,11 +349,19 @@ void eth_task(void *pvParameter)
         vTaskDelay(2000 / portTICK_PERIOD_MS);
 
         if (tcpip_adapter_get_ip_info(ESP_IF_ETH, &ip) == 0) {
-            ESP_LOGI(TAG, "\n~~~~~~~~~~~\n");
+#if (CONFIG_PHY_TLK110)
+            if (debug)
+                dump_tlk110_registers();
+#endif
+#if (CONFIG_PHY_LAN8720)
+            if (debug)
+                dump_lan8720_registers();
+#endif
```

Comment:
For code like the above it's preferable to define a single function name, something like `static void dump_phy_registers` and name both PHY-specific versions the same. The #ifdef guards will determine which one is compiled in.